test_log/src/main/java/org/corfudb/test_log/TestLog.java

```diff
@@ -0,0 +1,289 @@
+import com.google.common.hash.HashCode;
+import org.docopt.Docopt;
+
+public class TestLog {
+    private static int MetadataSize;
+    private static int LogHeaderSize;
+    private static final String USAGE =
+            "Usage:\n"
+                    + "\ttest_log [-d <show binary> | -e <address> | -E <address>]"
+                    + "\n"
+                    + "Options:\n"
+                    + " -d <show binary>, --display=<show binary>   Display log records (include binary data)\n"
+                    + " -e <address>, --erase=<address>             Erase log record\n"
+                    + " -E <address>, --erase_remain=<address>      Erase log records to end of file\n";
+
+    public static void main(String[] args) {
+        Map<String, Object> opts = new Docopt(USAGE).parse(args);
+        Operation op = new Operation(Operation.OperationType.REPORT, 0);
+        boolean use_output_file = false;
+        if (opts.get("--erase") != null) {
+            int addr = Integer.parseInt((String) opts.get("--erase"));
+            op = new Operation(Operation.OperationType.ERASE_ONE, addr);
+            use_output_file = true;
+        } else if (opts.get("--erase_remain") != null) {
+            int addr = Integer.parseInt((String) opts.get("--erase_remain"));
+            op = new Operation(Operation.OperationType.ERASE_REMAINING, addr);
+            use_output_file = true;
+        } else if (opts.get("--display") != null) {
+            int flag = Integer.parseInt((String) opts.get("--display"));
+            if (flag == 0) {
+                op = new Operation(Operation.OperationType.DISPLAY, 0);
+            } else {
+                op = new Operation(Operation.OperationType.DISPLAY_ALL, 0);
+            }
+        }
+        LogHeader lh = LogHeader.newBuilder()
+                .setVersion(10)
+                .setVerifyChecksum(true)
+                .build();
+        LogHeaderSize = lh.getSerializedSize();
+        Metadata md = Metadata.newBuilder()
+                .setChecksum(0x4d5ca4cc)
+                .setLength(lh.getSerializedSize())
+                .build();
+        MetadataSize = md.getSerializedSize();
+        if (false) {
+            System.out.println("md size is " + Integer.toString(md.getSerializedSize()));
+            System.out.println("lh size is " + Integer.toString(lh.getSerializedSize()));
+            ByteBuffer recordBuf = ByteBuffer.allocate(lh.getSerializedSize() + md.getSerializedSize());
+            //short delimiter = 0x4c45;
+            //recordBuf.putShort(delimiter);
+            recordBuf.put(md.toByteArray());
+            recordBuf.put(lh.toByteArray());
+            recordBuf.flip();  // reset pointers for write
+            String s = "";
+            while (recordBuf.hasRemaining()) {
+                byte b = recordBuf.get();
+                s += String.format("%02x", b);
+            }
+            System.out.println(s);
+        }
+
+        File f_in = new File("0.log");
```

Comment:
I guess that that the log file names would need to be inherited from the Corfu repo, and also be settable as command line parameters.